cluster-operator/controllers/vmc/vmc_controller.go

```
@@ -637,23 +682,43 @@ func (r *VerrazzanoManagedClusterReconciler) updateStatus(ctx context.Context, v
-	if err != nil {
-		return err
+// getCAPIClusterPhase returns the phase reported by the CAPI Cluster CR which is referenced by clusterRef.
+func (r *VerrazzanoManagedClusterReconciler) getCAPIClusterPhase(clusterRef *clustersv1alpha1.ClusterReference) (clustersv1alpha1.StateType, error) {
+	// Get the CAPI Cluster CR
+	cluster := &unstructured.Unstructured{}
+	cluster.SetGroupVersionKind(capi.GVKCAPICluster)
+	clusterNamespacedName := types.NamespacedName{
+		Name:      clusterRef.Name,
+		Namespace: clusterRef.Namespace,
+	}
+	if err := r.Get(context.TODO(), clusterNamespacedName, cluster); err != nil {
+		return "", err
 	}
 
-	// Replace the existing status conditions and state with the conditions generated from this reconcile
-	for _, genCondition := range vmc.Status.Conditions {
-		r.setStatusCondition(existingVMC, genCondition, genCondition.Type == clustersv1alpha1.ConditionManifestPushed)
+	// Get the state
+	phase, found, err := unstructured.NestedString(cluster.Object, "status", "phase")
+	if !found {
+		err = fmt.Errorf("could not find field status.phase field inside cluster %s: %v", clusterNamespacedName, err)
+		return "", err
+	}
+	if err != nil {
+		return "", err
 	}
-	existingVMC.Status.State = vmc.Status.State
-	existingVMC.Status.ArgoCDRegistration = vmc.Status.ArgoCDRegistration
 
-	r.log.Debugf("Updating Status of VMC %s: %v", vmc.Name, vmc.Status.Conditions)
-	return r.Status().Update(ctx, existingVMC)
+	// Validate that the CAPI Phase is a proper StateType for the VMC
+	switch state := clustersv1alpha1.StateType(phase); state {
+	case clustersv1alpha1.StatePending,
+		clustersv1alpha1.StateProvisioning,
+		clustersv1alpha1.StateProvisioned,
+		clustersv1alpha1.StateDeleting,
+		clustersv1alpha1.StateUnknown,
+		clustersv1alpha1.StateFailed:
+		return state, nil
+	default:
+		return "", fmt.Errorf("retrieved an invalid ClusterAPI Cluster phase of %s", state)
```

Consider defaulting to PhaseUnknown?

---

That does make sense to me. I can change that

---

I have changed this to default to `StateUnknown`, but upon manually testing this part of the code, I notice that returning an error results in hitting this line: https://github.com/verrazzano/verrazzano/pull/7191/files#diff-e43c1f5486c03e16914b7dcf76feb73c30b07dc75db40665e026de932d31cfc5R661, which leads to not updating the VMC state to `StateUnknown`.

I'm thinking of changing the order of `if` blocks in the part of code that I linked. Does anything pop out as wrong to you with this?
```
if capiClusterPhase != "" {
    vmc.Status.State = capiClusterPhase
}
if err != nil {
    return err
}
```

---

you don't need to return an error

---

Do you think the default case maybe needs a Progress message then? Saying that an invalid CAPI phase was retrieved?

---

I'm thinking now to do something like this under the default case:
```
r.log.Progressf("retrieved an invalid ClusterAPI Cluster phase of %s", state)
return clustersv1alpha1.StateUnknown, nil
```

---

sure

---

I pushed the commit for this change and manually verified that it works as expected.